insecure/corruptible/network.go

```diff
@@ -0,0 +1,370 @@
+	approvalHasher hash.Hasher
+}
+
+var _ flownet.Network = &Network{}
+
+func NewCorruptibleNetwork(
+	logger zerolog.Logger,
+	chainId flow.ChainID,
+	address string,
+	me module.Local,
+	codec flownet.Codec,
+	flowNetwork flownet.Network,
+	conduitFactory insecure.CorruptibleConduitFactory) (*Network, error) {
+	if chainId != flow.BftTestnet {
+		panic("illegal chain id for using corruptible network")
+	}
+
+	corruptibleNetwork := &Network{
+		codec:          codec,
+		me:             me,
+		conduitFactory: conduitFactory,
+		flowNetwork:    flowNetwork,
+		logger:         logger.With().Str("component", "corruptible-network").Logger(),
+		receiptHasher:  utils.NewExecutionReceiptHasher(),
+		spockHasher:    utils.NewSPOCKHasher(),
+		approvalHasher: verutils.NewResultApprovalHasher(),
+	}
+
+	err := corruptibleNetwork.conduitFactory.RegisterEgressController(corruptibleNetwork)
+	if err != nil {
+		return nil, fmt.Errorf("could not register egress controller on conduit factory: %w", err)
+	}
+	corruptibleNetwork.ComponentManager = component.NewComponentManagerBuilder().
+		AddWorker(func(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+			corruptibleNetwork.flowNetwork.Start(ctx)
+			<-corruptibleNetwork.flowNetwork.Ready()
+
+			ready()
+
+			<-corruptibleNetwork.flowNetwork.Done()
+		}).
+		AddWorker(func(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+			corruptibleNetwork.start(ctx, address)
+			ready()
+
+			<-ctx.Done()
+			corruptibleNetwork.stop()
+
+		}).Build()
+
+	return corruptibleNetwork, nil
+}
+
+// Register serves as the typical network registration of the given message processor on the channel.
+// Except, it first wraps the given processor around a corruptible message processor, and then
+// registers the corruptible message processor to the original flow network.
+func (n *Network) Register(channel channels.Channel, messageProcessor flownet.MessageProcessor) (flownet.Conduit, error) {
+	corruptibleProcessor := NewCorruptibleMessageProcessor(n.logger, messageProcessor)
+	// TODO: we can dissolve CCF and instead have a decorator pattern to turn a conduit into
+	// a corrupted one?
+	conduit, err := n.flowNetwork.Register(channel, corruptibleProcessor)
+	if err != nil {
+		return nil, fmt.Errorf("could not register corruptible message processor on channel: %s, %w", channel, err)
+	}
+	return conduit, nil
+}
+
+func (n *Network) RegisterBlobService(channel channels.Channel, store datastore.Batching, opts ...flownet.BlobServiceOption) (flownet.BlobService,
+	error) {
+	return n.flowNetwork.RegisterBlobService(channel, store, opts...)
+}
+
+func (n *Network) RegisterPingService(pingProtocolID protocol.ID, pingInfoProvider flownet.PingInfoProvider) (flownet.PingService, error) {
+	return n.flowNetwork.RegisterPingService(pingProtocolID, pingInfoProvider)
+}
+
+func (n *Network) ProcessAttackerMessage(stream insecure.CorruptibleConduitFactory_ProcessAttackerMessageServer) error {
```

Comment:
Can we add a comment here about this being the main place messages are processed by the corrupted network and how it's called by gRPC?

Reply:
https://github.com/onflow/flow-go/pull/2807/commits/1df4164215782d40e304157a533259d36bc53236